pkg/tenant/db/errors.go

```diff
@@ -49,22 +54,44 @@ var (
 	ErrTopicNameTooLong          = errors.New("topic name is too long")
 )
 
+// ValidationError represents a validation error for a specific field and may contain
+// an index if the field is a slice.
 type ValidationError struct {
-	err error
+	Field string
+	Err   error
+	Index int
 }
 
-func invalid(err error) *ValidationError {
-	return &ValidationError{err}
+func validationError(field string, err error) *ValidationError {
+	return &ValidationError{
+		Field: field,
+		Err:   err,
+	}
+}
+
+func (e *ValidationError) AtIndex(index int) *ValidationError {
+	e.Index = index
+	return e
 }
 
 func (e *ValidationError) Error() string {
-	return fmt.Sprintf("validation error: %s", e.err)
+	return fmt.Sprintf("invalid field '%s': %s", e.Field, e.Err)
```

Comment:
Should we handle the case where there is no field or the case where there is an index in the string?